### ISSUE TITLE
qrcode.makeCode() still has Code Length OverFlow Error

### ISSUE THREAD

**Issue by suxiaogang** [KeeeX/qrcodejs#1]
Come form [https://github.com/davidshimjs/qrcodejs/issues/78](url) and firstly thanks for you awesome fork for the original repo. 
it seems that `qrcode.makeCode()` still has this issue.  please do a further job, thank you.

**Comment by fladna9** [comment]
Hello @suxiaogang. 
I need more information about your problem. What is the length (eventually content if not private) of the QR Code you're trying to generate ?
Did you try any other correctLevel parameter ?

**Comment by suxiaogang** [comment]
emmmmm.... after investigation this might be related to character encoding ... 
actually I'm try to qrcode some string which contains Chinese 
```javascript
var QRCodeBox = new QRCode(document.getElementById("qr"), {
    width: 200,
    height: 200,
    correctLevel: QRCode.CorrectLevel.H
});
var str = "ed2k://|file|【电影家园www.idyjy.com下载】寂静之地.HD韩版中英双字.mp4|1129787457|AD3BD4384712D52D2BB09A150AF64D55|h=BMZZS7VBJ4ZAROGOU56245LCBBO6DRO7|/";
QRCodeBox.makeCode(str);
```

@fladna9 well you can close this issue if it is confirmed to be a encoding problem.

**Comment by fladna9** [comment]
Strange. Got this on Chrome Version 66.0.3359.181 (Official build) (64 bits) on macOS, no warnings/errors in JS console...
Will try in Firefox and Safari.
![telechargement](https://user-images.githubusercontent.com/18591264/40470223-2ef911a4-5f33-11e8-9608-d849f2a402f1.png)

**Comment by suxiaogang** [comment]
Chrome/65.0.3325.181 & Mac OS X 10_13_4, everytime

```html
<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8" />
</head>
<body class="chrome-popup">
<div style="width:100%;margin:35px;" id="qr"></div>
<script src="http://cdn.bootcss.com/jquery/1.12.4/jquery.min.js"></script>
<!--replace below with your forked file-->
<script src="https://raw.githubusercontent.com/KeeeX/qrcodejs/master/qrcode.min.js"></script>
</body>
<script>
$(document).ready(function() {
  var QRCodeBox = new QRCode(document.getElementById("qr"), {
      width: 200,
      height: 200,
      correctLevel: QRCode.CorrectLevel.H
  });
  var str = "ed2k://|file|【电影家园www.idyjy.com下载】寂静之地.HD韩版中英双字.mp4|1129787457|AD3BD4384712D52D2BB09A150AF64D55|h=BMZZS7VBJ4ZAROGOU56245LCBBO6DRO7|/";
  QRCodeBox.makeCode(str);
});
</script>
</html>
```

```log
qrcode.min.js:1 Uncaught Error: code length overflow. (3004>1440)
    at Function.b.createData (qrcode.min.js:1)
    at b.makeImpl (qrcode.min.js:1)
    at b.getBestMaskPattern (qrcode.min.js:1)
    at b.make (qrcode.min.js:1)
    at QRCode.makeCode (qrcode.min.js:1)
    at HTMLDocument.<anonymous> (test.html:22)
    at i (jquery.min.js:2)
    at Object.fireWith [as resolveWith] (jquery.min.js:2)
    at Function.ready (jquery.min.js:2)
    at HTMLDocument.K (jquery.min.js:2)
```

**Comment by fladna9** [comment]
Ok, problem is the `qrcode.min.js` file.
It works fine on `qrcode.js`

Tested on Chrome, Safari, Firefox, and had no problem so far. Below is the code of the Testpage i used (just edited `index.html` with your JS code).

![capture d ecran 2018-05-24 a 09 18 49](https://user-images.githubusercontent.com/18591264/40471409-e6f6fc82-5f36-11e8-9bdf-9b49b0d4c627.png)

```
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="ko" lang="ko">
<head>
	<title>Cross-Browser QRCode generator for Javascript</title>
	<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
	<meta name="viewport" content="width=device-width,initial-scale=1,user-scalable=no" />
	<script type="text/javascript" src="jquery.min.js"></script>
	<script type="text/javascript" src="qrcode.js"></script>
</head>
<body>
	<div id="qrcode" style="width:100px; height:100px; margin-top:15px;"></div>

	<script type="text/javascript">
		var QRCodeBox = new QRCode(document.getElementById("qrcode"), {
			width: 200,
			height: 200,
			correctLevel: QRCode.CorrectLevel.H
		});
		var str = "ed2k://|file|【电影家园www.idyjy.com下载】寂静之地.HD韩版中英双字.mp4|1129787457|AD3BD4384712D52D2BB09A150AF64D55|h=BMZZS7VBJ4ZAROGOU56245LCBBO6DRO7|/";
		QRCodeBox.makeCode(str);
	</script>
</body>
```

Will redo a min.js as soon as i have a bit of time for this.
Thanks for the bug submission.

In the mean time, i highly recommend you to use the non-minified version, or to minify it yourself...
I'm not sure i'll have time today to do this. ;-)

Have a good day !

**Comment by fladna9** [comment]
I will leave this Open until a fixed `qrcode.min.js` will be uploaded.

**Comment by suxiaogang** [comment]
nice job dude

**Comment by fladna9** [comment]
Fixed in df9405a, closing issue.
